website/content/docs/concepts/transform.mdx

```diff
@@ -2,36 +2,125 @@
+
+<ImageConfig hideBorder>
+
+![Format preserving encryption](/img/vault-encoded-text.jpg)
+
+</ImageConfig>
+
+Maintaining a similar data structure for encoded sensitive data allows for flexible storage without requiring extensive database changes. For instance, consider a scenario with a substantial customer database containing billing information that requires protection. By ensuring that the encoded ciphertext maintains a similar format, you can store the ciphertext back into your original systems with minimal changes to the database. 
+
+FPE, in contrast to conventional tokenization methods, produces encrypted text that preserves the original data's structure and format while ensuring the security of the encoded values. By utilizing FPE in Vault, users can create custom transformation templates using regular expressions. FPE enables users to specify the values they wish to transform and enforces a schema on the encoded value. In addition, there are pre-existing templates, such as those for credit card numbers and US social security numbers, readily available to help you get started quickly.
+
+To ensure the security of the encoded ciphertexts, Vault utilizes the [FF3-1](https://csrc.nist.gov/publications/detail/sp/800-38g/rev-1/draft) encryption algorithm vetted by the National Institute of Standards and Technology (NIST). The FF3-1 algorithm is updated and tested to protect against specific types of attacks and potential future threats from future supercomputers.
+
+FPE transformation is stateless, meaning that Vault does not store the protected secret; rather it protects only the encryption key material necessary to decrypt the secret's ciphertext. Storing only the necessary material to decrypt ensures maximum performance for encoding and decoding operations while also minimizing the possibility of exposure of that secret.
+
+#### Learn more about transform
+
+- [Transform secrets engine tutorial](/vault/tutorials/adp/transform)
+- [Encrypting data with Transform secrets engine](/vault/tutorials/adp/transform-code-example)
+- [Transform secrets engine API](/vault/api-docs/secret/transform)
+
+### Data masking
+
+Data masking is a one-way transformation that performs the replacement of matched characters on the input value with a desired character. This form of transformation is non-reversible and thus does not support retrieving the original value back using the decode operation.
+
+Utilize data masking when showing or printing sensitive data without full readability. For instance, online banking systems commonly mask users' account numbers to prevent potential security breaches by bad actors who might be observing the screen, unbeknownst to the account owner.
```

Comment:
```suggestion
Data masking is a one-way transformation that replaces characters on the input
value with a predefined translation character.

<Warning title="Use with caution">

  Masking is non-reversible. We do not recommend masking for situations where
  you need to retrieve and decode the original value.

</Warning>

Data masking is a good solution when you need to show or print sensitive data
without full readability. For example, masking a a bank account number in an
online banking portal to prevent potential security breaches from bad actors who
might be observing the screen.
```
Style correction: write in active voice, avoid possessives